web/modules/custom/employees/src/Plugin/Action/ChangeContactAction.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\employees\Plugin\Action;
+
+use Drupal\views_bulk_operations\Action\ViewsBulkOperationsActionBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Publish content
+ *
+ * @Action(
+ *   id = "employees_change_contact_action",
+ *   label = @Translation("Change contact (custom action)"),
+ *   type = "node",
+ *   confirm = FALSE,
+ * )
+ */
+class ChangeContactAction extends ViewsBulkOperationsActionBase {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+    if ($entity === NULL || !$entity->hasField('field_contact')) {
+      return $this->t('Invalid entity or configuration.');
+    }
+  
+    $contacts = $entity->get('field_contact');
+    
+    // Look through all contacts to find the old contact and replace it with the new contact
+    foreach ($contacts as $contact) {
+      $user_id = $contact->getValue()['target_id'];
+      if ($user_id == $this->configuration['old_user_id']) {
+        $contact->setValue($this->configuration['new_user_id']);
+
+        $entity->save();
```

Comment:
What do you think about adding a revision when the contact is changed/removed?

Reply:
Good idea, I'll look into that.

Reply:
Done. Both actions now record a revision log showing what was changed.